java/libphonenumber/src/com/google/i18n/phonenumbers/PhoneNumberUtil.java

```diff
@@ -320,15 +326,16 @@
     // hardly ever used to indicate this.
     String singleExtnSymbolsForParsing = ",;" + singleExtnSymbolsForMatching;
 
-    EXTN_PATTERNS_FOR_PARSING = createExtnPattern(singleExtnSymbolsForParsing);
-    EXTN_PATTERNS_FOR_MATCHING = createExtnPattern(singleExtnSymbolsForMatching);
+    EXTN_PATTERNS_FOR_PARSING = createExtnPattern(singleExtnSymbolsForParsing, CAPTURING_EXTN_DIGITS_AND_COMMANDS);
+    EXTN_PATTERNS_FOR_MATCHING = createExtnPattern(singleExtnSymbolsForMatching, CAPTURING_EXTN_DIGITS);
   }
 
   /**
    * Helper initialiser method to create the regular-expression pattern to match extensions,
    * allowing the one-char extension symbols provided by {@code singleExtnSymbols}.
+   * {@code extensionPattern} allows to specify the extension capturing pattern.
    */
-  private static String createExtnPattern(String singleExtnSymbols) {
+  private static String createExtnPattern(String singleExtnSymbols, String extensionPattern) {
```

Comment:
Instead of extensionPattern I'd call it capturingExtnDigits because the comment below explains that there are three regular expressions involved and arguably those are "extension patterns" too (e.g. RFC3966_EXTN_PREFIX).